tests/Microsoft.Identity.Test.Unit/CoreTests/InstanceTests/InstanceDiscoveryManagerTests.cs

```diff
@@ -42,14 +42,19 @@ public override void TestInitialize()
             _networkCacheMetadataProvider = Substitute.For<INetworkCacheMetadataProvider>();
             _userMetadataProvider = Substitute.For<IUserMetadataProvider>();
 
+            InitializeTestObjects();
+        }
+
+        private void InitializeTestObjects(bool isInstanceDiscoveryEnabled = true)
```

Comment:
Can we have a simple unit test around the end to end, similar to `ConfidentialClientUsingSecretNoCacheProvidedTestAsync`: 

- set this flag to true
- remove the mock `httpManager.AddInstanceDiscoveryMockHandler();`